chain/client/tests/catching_up.rs

```diff
@@ -641,4 +641,153 @@ mod tests {
+                                }
+                            }
+                            if let NetworkRequests::PartialEncodedChunkMessage {
+                                account_id, ..
+                            } = msg
+                            {
+                                if account_id == "test1.2" {
+                                    return (NetworkResponses::NoResponse, false);
+                                }
+                            }
+                            if let NetworkRequests::Block { block } = msg {
+                                println!(
+                                    "HOLDING PHASE, HEIGHT = {:?}",
+                                    block.header.inner_lite.height
+                                );
+                                if block.header.inner_lite.height >= height_release {
+                                    *phase = OrphansPhases::ValidatingOrphans;
+                                }
+                            }
+                        }
+                        OrphansPhases::ValidatingOrphans => {
+                            if let NetworkRequests::Block { block } = msg {
+                                println!(
+                                    "VALIDATING PHASE, HEIGHT = {:?}",
+                                    block.header.inner_lite.height
+                                );
+                                assert!(block.header.inner_lite.height >= height_release);
+                                assert!(block.header.inner_lite.height <= height_release + 10);
+                                if block.header.inner_lite.height == height_release + 10 {
+                                    System::current().stop();
+                                }
+                                if block.header.inner_lite.height >= height_release + 5 {
+                                    for i in 0..16 {
+                                        actix::spawn(
+                                            connectors1.write().unwrap()[i]
+                                                .1
+                                                .send(Query::new(
+                                                    "account/".to_string() + &account_to,
+                                                    vec![],
+                                                ))
+                                                .then(move |res| {
+                                                    let res_inner = res.unwrap();
+                                                    if let Ok(Some(query_response)) = res_inner {
+                                                        if let ViewAccount(view_account_result) =
+                                                            query_response
+                                                        {
+                                                            assert_eq!(
```

Comment:
Add a hashset of all validators for whom this check was triggered at least once, and do not pass the test unless the hashset has all of them.